build.gradle

```diff
@@ -136,76 +132,49 @@ sourceSets {
-def sampleJSONPOJOs = [
-        // alpaca/
-        ['path': 'alpaca/account/', 'sourceType': 'json'],
-        ['path': 'alpaca/asset/', 'sourceType': 'json'],
-        ['path': 'alpaca/bar/', 'sourceType': 'json'],
-        ['path': 'alpaca/calendar/', 'sourceType': 'json'],
-        ['path': 'alpaca/clock/', 'sourceType': 'json'],
-        ['path': 'alpaca/order/', 'sourceType': 'json'],
-        ['path': 'alpaca/position/', 'sourceType': 'json'],
-
-        // polygon/
-        ['path': 'polygon/aggregate/', 'sourceType': 'json'],
-        ['path': 'polygon/historic/quotes/', 'sourceType': 'json'],
-        ['path': 'polygon/historic/trades/', 'sourceType': 'json'],
-        ['path': 'polygon/last/', 'sourceType': 'json'],
-        ['path': 'polygon/meta/', 'sourceType': 'json'],
-        ['path': 'polygon/open_close/', 'sourceType': 'json'],
-        ['path': 'polygon/reference/', 'sourceType': 'json'],
-        ['path': 'polygon/snapshot/', 'sourceType': 'json'],
-        ['path': 'polygon/websocket/', 'sourceType': 'json']
-]
-
 task generatePOJOs() {
     final def targetDirectory = "${project.buildDir}/generated-sources/samplejson"
-    final def jsonSourceDirectory = 'sample_json/'
-    def first = true
-
-    sampleJSONPOJOs.each { propertyMap ->
-        def path = propertyMap.get('path')
-        def sourceType = propertyMap.get('sourceType')
-        def targetPackage = 'io.github.mainstringargs.domain.' +
-                path.substring(0, path.lastIndexOf('/')).replace('/', '.')
-
-        def jsonToPOJOTask = tasks.create('jsonToPOJOTask-' + targetPackage, GradleBuild) {
-            startParameter.projectProperties = ['source'         : jsonSourceDirectory + path,
-                                                'sourceType'     : sourceType,
-                                                'targetPackage'  : targetPackage,
-                                                'targetDirectory': targetDirectory]
-            buildFile = 'sampleJson2Pojo.gradle'
-            tasks = ['generateJsonSchema2Pojo']
-        }
+    final def jsonSourceDirectory = "${project.projectDir}/sample_json/"
+    final def jsonPackageNameStart = "io.github.mainstringargs.domain."
+
+    // Clear all generated sources
+    delete targetDirectory
+
+    // Loop through all files in sample JSON file tree
+    file(jsonSourceDirectory).eachFileRecurse { file ->
+        if (file.getName().endsWith(".json")) {
+            def startPackageIndex = file.getAbsolutePath().indexOf(jsonSourceDirectory) + jsonSourceDirectory.length()
+            def targetPackage = jsonPackageNameStart + file.getAbsolutePath()
+                    .replace('/' + file.getName(), '')
+                    .substring(startPackageIndex)
+                    .replace('/', '.').replace('-', '').replace('_', '')
+                    .toLowerCase()
+
+            def jsonToPOJOTask = tasks.create('json2POJOTask-' + targetPackage + "." + file.getName(), GradleBuild) {
```

Comment:
I'm getting some errors when doing gradlew build -- and I'm trying to decide if it's something with my environment or a difference between Linux and Windows.   It looks like it is putting in slashes where it should be putting in periods when it is generating sources from the Json:

![image](https://user-images.githubusercontent.com/7042674/69974250-7e2cf600-14ea-11ea-85eb-238de0660dc2.png)

![image](https://user-images.githubusercontent.com/7042674/69974315-a157a580-14ea-11ea-923e-684c48e7a855.png)

Have you seen this or any thoughts on what may be happening?

Reply:
Ahh I see what's happening. I develop on Mac so I neglected to replace the Windows file separator character when creating a unique task name. I'll fix this later today.

Reply:
Fixed! I've simply replaced `/` with `File.separator`.